pkg/endpoint/endpoint.go

```diff
@@ -1564,9 +1565,16 @@ func (e *Endpoint) SetStateLocked(toState, reason string) bool {
 		// No valid transitions, as disconnected is a terminal state for the endpoint.
 	case StateWaitingToRegenerate:
 		switch toState {
-		// Note that transitions to waiting-to-regenerate state
+		// Note that transitions to StateWaitingToRegenerate are not allowed,
+		// as callers of this function enqueue regenerations if 'true' is
+		// returned. We don't want to return 'true' for the case of
+		// transitioning to StateWaitingToRegenerate, as this means that a
+		// regeneration is already queued up. Callers would then queue up
+		// another unneeded regeneration, which is undesired.
 		case StateWaitingForIdentity, StateDisconnecting, StateRestoring:
 			goto OKState
+		default:
```

Comment:
I would prefer this be `case StateWaitingToRegenerate:` instead.

Reply:
Wouldn't be simpler if we do only:
```golang
		// Note that transitions to StateWaitingToRegenerate are not allowed,
		// as callers of this function enqueue regenerations if 'true' is
		// returned. We don't want to return 'true' for the case of
		// transitioning to StateWaitingToRegenerate, as this means that a
		// regeneration is already queued up. Callers would then queue up
		// another unneeded regeneration, which is undesired.
		case StateWaitingToRegenerate:
			return false
```

Reply:
Done!